File: app/src/main/java/in/kashewdevelopers/todo/adapters/TaskViewHolder.java
package in.kashewdevelopers.todo.adapters;

import android.view.View;

import androidx.recyclerview.widget.RecyclerView;

import in.kashewdevelopers.todo.data_container.TaskDataContainer;
import in.kashewdevelopers.todo.databinding.TaskListItemLayoutBinding;

public class TaskViewHolder extends RecyclerView.ViewHolder implements View.OnClickListener, View.OnLongClickListener {

    public View view;
    public TaskListItemLayoutBinding binding;
    public TaskDataContainer data;

    public TaskViewHolder(View view) {
        super(view);
        this.view = view;
        binding = TaskListItemLayoutBinding.bind(view);

        binding.root.setOnClickListener(this);
        binding.root.setOnLongClickListener(this);

        binding.checkbox.setOnClickListener(this);
    }

    @Override
    public void onClick(View view) {
        view.setPressed(true);
        onClick(false);
    }

    @Override
    public boolean onLongClick(View v) {
        onClick(true);
        return true;
    }


    public void onClick(boolean showDetails) {
        if (onItemClickListener != null) {
            onItemClickListener.onItemClickListener(this, showDetails);
        }
    }


    // callbacks
    public interface OnItemClickListener {
        void onItemClickListener(TaskViewHolder holder, boolean showDetail);
    }

    private OnItemClickListener onItemClickListener;

    public TaskViewHolder setOnItemClickListener(OnItemClickListener onItemClickListener) {
        this.onItemClickListener = onItemClickListener;
        return this;
    }

}

File: app/src/main/java/in/kashewdevelopers/todo/adapters/TaskViewHolder.kt
package `in`.kashewdevelopers.todo.adapters

import `in`.kashewdevelopers.todo.data_container.TaskDataContainer
import `in`.kashewdevelopers.todo.databinding.TaskListItemLayoutBinding
import android.view.View
import androidx.recyclerview.widget.RecyclerView

class TaskViewHolder(view: View) : RecyclerView.ViewHolder(view), View.OnClickListener, View.OnLongClickListener {

    val binding = TaskListItemLayoutBinding.bind(view)
    lateinit var data: TaskDataContainer

    init {
        binding.root.setOnClickListener(this)
        binding.root.setOnLongClickListener(this)

        binding.checkbox.setOnClickListener(this)
    }

    override fun onClick(view: View) {
        view.isPressed = true
        onClick(false)
    }

    override fun onLongClick(view: View): Boolean {
        onClick(true)
        return true
    }

    private fun onClick(showDetails: Boolean) {
        onItemClickListener?.onItemClickListener(this, showDetails)
    }


    // callbacks
    interface OnItemClickListener {
        fun onItemClickListener(holder: TaskViewHolder, showDetail: Boolean)
    }

    private var onItemClickListener: OnItemClickListener? = null

    fun setOnItemClickListener(onItemClickListener: OnItemClickListener?): TaskViewHolder {
        this.onItemClickListener = onItemClickListener
        return this
    }

}